src/PluginBase/PluginContext.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <PluginBase/RaPluginBase.hpp>
+#include <QObject>
+
+namespace Ra {
+namespace Engine {
+class RadiumEngine;
+}
+namespace GuiBase {
+class SelectionManager;
+}
+namespace Gui {
+class PickingManager;
+class Viewer;
+} // namespace Gui
+
+namespace Plugins {
+
+/// Data passed to the plugin constructor.
+class RA_PLUGINBASE_API Context : public QObject
+{
+    Q_OBJECT
+
+  public:
+    explicit Context( QObject* parent = nullptr ) : QObject( parent ) {}
+
+    Engine::RadiumEngine* m_engine{nullptr};
+    GuiBase::SelectionManager* m_selectionManager{nullptr};
+    Gui::PickingManager* m_pickingManager{nullptr};
+    Gui::Viewer* m_viewer{nullptr};
+    std::string m_exportDir{"."};
+
+  signals:
+    /*!
+     * \brief enable continuous rendering update
+     * \see BaseApplication::setContinuousUpdate
+     *
+     * Usage: in your plugin, declare signals, and connect them to the PluginContext signals for
+     * forwarding:
+     * \code
+     * class MyPlugin : public QObject, Ra::Plugins::RadiumPluginInterface
+     * {
+     *   void MyPlugin::registerPlugin( const Ra::Plugins::Context& context ) {
+     *     connect( this, &MyPlugin::askForUpdate, &context, &Ra::Plugins::Context::askForUpdate );
+     *     connect( this, &MyPlugin::setContinuousUpdate, &context,
+     * &Ra::Plugins::Context::setContinuousUpdate );
+     *   }
+     *
+     * signals:
+     *   void setContinuousUpdate( bool b );
+     *   void askForUpdate();
```

Comment:
Shall we have these in the `RadiumPluginInterface` ?

Reply:
No, you cannot have signal and slots in an interface, because you need MOC to parse the class.